neo4j/src/main/java/org/hibernate/ogm/datastore/neo4j/remote/common/dialect/impl/RemoteNeo4jAssociationSnapshot.java

```diff
@@ -29,6 +31,11 @@ public RemoteNeo4jAssociationSnapshot(Map<RowKey, Tuple> tuples) {
 	@Override
 	public Tuple get(RowKey rowKey) {
 		Tuple tuple = tuples.get( rowKey );
+		for ( Entry<RowKey, Tuple> entry : tuples.entrySet() ) {
+			if ( RemoteNeo4jHelper.matches( entry.getKey(), rowKey ) ) {
+				return entry.getValue();
+			}
+		}
```

Comment:
The logic is weird here. Maybe you should directly return the tuple from the first line if not null? That would work in the simple case and if you don't find it you execute the fallback logic.

At least, that's how I understand what you're trying to do but I may be wrong.